internal/telemetry/tracker.go

```diff
@@ -26,15 +33,32 @@ func (tracker *stdoutTracker) Track(event event) {
 	)
 }
 
-type segmentTracker struct{}
+type segmentTracker struct {
+	client analytics.Client
+	logger *log.Logger
+}
+
+func newSegmentTracker(logger *log.Logger) Tracker {
+	if len(segmentWriteKey) == 0 {
+		log.Print("unable to connect to Segment due to missing key, CLI telemetry will be disabled")
+		return &noopTracker{}
+	}
+	client := analytics.New(segmentWriteKey)
+	return &segmentTracker{client, logger}
+}
 
-// TODO(REALMC-7243): use Segment sdk to send events through client
 func (tracker *segmentTracker) Track(event event) {
-	fmt.Printf(
-		"%s UTC TELEM %s: %s%v\n",
-		event.time.In(time.UTC).Format("15:04:05"),
-		event.command,
-		event.eventType,
-		event.data,
-	)
+	properties := make(map[string]interface{}, len(event.data))
```

Comment:
btw we still want to be adding `event.command` and `event.executionID` as properties here (see my previous comment re: `"cmd"` and `"xid"` event data keys

Reply:
Is there a reason why `event.Data` is an array? Can it just be `"err"`? It seems strange that `cmd` and `xid` are taken from the event at a different level than `err` is (on the `event` object itself vs within the `data` property)

Reply:
if you look at the signature for `(*telemetry.Service).TrackEvent`, you'll see the minimal piece of information needed by the calling code is the `EventType` to track on, and it optionally supports any number of additional `EventData` tuples/structs/objects

for the "start" and "complete" events, we don't really care about any extra details, but on the "error" event we leverage that ability to include an error message

otherwise, the properties that we want to be able to see in every event tracked are provided at the time when we create the `telemetry.Service` (user id, command name).  those are known ahead of time, so we avoid passing them in every time to the `start`/`error`/`complete` stages

Reply:
Should I add `cmd` and `xid` as `EventDataKey`s ?

Reply:
we can add them as `EventDataKey`, but no need to export them:

```go
// set of event data keys
const (
	eventDataKeyCommand     EventDataKey = "cmd"
	eventDataKeyExecutionID EventDataKey = "xid"

	EventDataKeyErr EventDataKey = "err"
)
```